plugin/trino-hive/src/main/java/io/trino/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -529,7 +529,7 @@ private ListenableFuture<Void> loadPartition(HivePartitionMetadata partition)
             }
 
             JobConf jobConf = toJobConf(configuration);
-            FileInputFormat.setInputPaths(jobConf, path);
+            hdfsEnvironment.doAs(hdfsContext.getIdentity(), () -> FileInputFormat.setInputPaths(jobConf, path));
```

Comment:
Why is this needed? The method seems to only modify `JobConf`, not do any IO. If we inlined the method and simplified it, the result would be a single line:
```java
jobConf.set("mapreduce.input.fileinputformat.inputdir", StringUtils.escapeString(path.toString()));